include/method_enums.hpp

```diff
@@ -1,5 +1,14 @@
 #pragma once
 
+//!\brief An enum for the different clustering methods.
+enum detecting_methods
+{
+    cigar_string = 1,
+    split_read = 2,
+    read_pairs = 3,
+    read_depth = 4
+};
```

Comment:
## If there may be more detecting methods in the future, you could think about doing this:

```suggestion
//!\brief An enum for the different clustering methods.
enum detecting_methods
{
    cigar_string = 0,
    split_read = 1,
    read_pairs = 2,
    read_depth = 3,
    //!\cond
    // ATTENTION: Must always be the last item; will be used to determine the number of ids.
    SIZE //!< Determines the size of the enum.
    //!\endcond
};
```

Then you know the size via `detecting_methods::SIZE`,
there could for example be a test that tests that  `args.methods.size() == detecting_methods::SIZE`, i.e. the default method selection uses all methods.

Reply:
@eldariont does this sound usefull to you?

Reply:
Besides the test that Enrico suggested I cannot think of other applications of SIZE (can you?). It does not hurt to include the size though. However, I would add the suggested test already to this PR.

Two technical questions for my understanding: Why does this SIZE work? Is it because it get the value 4 by default and thus returns the size? And is there no other way of getting the size of an enum without setting SIZE explicitly? 🙂

Reply:
> Besides the test that Enrico suggested I cannot think of other applications of SIZE (can you?). It does not hurt to include the size though. However, I would add the suggested test already to this PR.

I also don't see another application, but if the default shall be "all methods", then it would probably be good to test this somewhere. Seems like something that is easy to forget. There may be another application regarding allowing certain combinations, e.g. creating a table of allowed combinations (which would require a known size).

> Two technical questions for my understanding: Why does this SIZE work? Is it because it gets the value 4 by default and thus returns the size? And is there no other way of getting the size of an enum without setting SIZE explicitly? 🙂

`SIZE` only works because we do the standard enumeration (0,1,2,3,4,5,...); otherwise there is no (? at least no simple) way to get the number of entries in an enum.